src/util/cfdict_helpers.c

```diff
@@ -0,0 +1,46 @@
+#include "cfdict_helpers.h"
+
+const void* ffCfDictGetValue(CFMutableDictionaryRef dict, const char* str)
+{
+    CFStringRef key = CFStringCreateWithCStringNoCopy(NULL, str, kCFStringEncodingASCII, kCFAllocatorNull);
+    return CFDictionaryGetValue(dict, key);
+}
+
+FFstrbuf ffCfDictGetString(CFMutableDictionaryRef dict, const char* str)
+{
+    CFStringRef cfStr = (CFStringRef)ffCfDictGetValue(dict, str);
+    uint32_t length = (uint32_t)CFStringGetLength(cfStr);
+    FFstrbuf ffStr;
+    ffStrbufInitA(&ffStr, length + 1);
+    if(CFStringGetCString(cfStr, ffStr.chars, length + 1, kCFStringEncodingASCII))
+    {
+        ffStr.length = length;
+        // CFStringGetCString ensures the buffer is NUL terminated
+        // https://developer.apple.com/documentation/corefoundation/1542721-cfstringgetcstring
+    }
+    return ffStr;
+}
+
+int ffCfDictGetBool(CFMutableDictionaryRef dict, const char* str)
+{
+    CFBooleanRef cfNum = (CFBooleanRef)ffCfDictGetValue(dict, str);
+    return CFBooleanGetValue(cfNum);
+}
```

Comment:
There is no error handling here. What happens if the key doesn't exist? Is false returned?